### ISSUE TITLE
composition of SQL functions

### ISSUE THREAD

**Issue by pkoppstein** [orientechnologies/orientdb#5500]
At present, one has to write:

```
select distinct(@this)
from (select expand(  unionall($a,$b) )  
          let $a = (select from #11:0),
               $b = (select from #11:0) )
```

whereas one would expect to be able to write:

```
select distinct(expand($this))
from (select unionall( (select from #11:0), (select from #11:0) ) )
```

This might be an ER but then again it's hard to imagine functions that cannot be composed.

**Comment by luigidellaquila** [comment]
HI @pkoppstein 

I'm flagging it as an enhancement, it will be implemented in 3.0

**Comment by lvca** [comment]
Implemented using DISTINCT keyword in 3.0.
